group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -495,6 +502,7 @@ private void transitionTo(
 
                 case ACTIVE:
                     state = CoordinatorState.ACTIVE;
+                    highWatermarklistener = new HighWatermarkListener();
```

Comment:
did we change this because we want to not accidentally retain state from the previous load?

Reply:
That’s right. We always start with a clean state.